src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -162,6 +187,33 @@ public static function get_core_data() {
 		return static::$core;
 	}
 
+	/**
+	 * Checks whether the registered blocks are the same
+	 * as the ones we already processed for this origin.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param $origin One of 'core', 'blocks', 'theme', 'user'. Source of data we cache blocks for.
+	 *
+	 * @return boolean True on success, false otherwise.
+	 */
+	protected static function has_same_registered_blocks( $origin ) {
+		$registry = WP_Block_Type_Registry::get_instance();
+		$blocks   = $registry->get_all_registered();
+
+		// Is there metadata for all currently registered blocks?
+		$block_diff = array_diff_key( $blocks, static::$blocks_cache[ $origin ] );
```

Comment:
What if the `$origin` is not one of the expected values? An error will happen. Adding a guard clause to the start of this method will protect against this while avoiding unnecessary processing.

```suggestion
	protected static function has_same_registered_blocks( $origin ) {
		// Bail out if the origin is invalid.
		if ( ! isset( static::$blocks_cache[ $origin ] ) {
			return false;
		}

		$registry = WP_Block_Type_Registry::get_instance();
		$blocks   = $registry->get_all_registered();

		// Is there metadata for all currently registered blocks?
		$block_diff = array_diff_key( $blocks, static::$blocks_cache[ $origin ] );
```